server/src/main/java/org/tctalent/server/service/db/impl/CandidateServiceImpl.java

```diff
@@ -3035,4 +3035,41 @@ public Page<Candidate> fetchCandidatesWithChat(
         }
     }
 
+    @Override
+    public void reassignSavedListCandidates(SavedList savedList, int partnerId) {
+        Partner newPartner = partnerService.getPartner(partnerId);
+
+        SavedListGetRequest request = new SavedListGetRequest();
+
+        Page<Candidate> candidatePage = getSavedListCandidates(savedList, request);
+
+        int totalPagesToProcess = candidatePage.getTotalPages();
+        int pagesProcessed = 0;
+
+        while (pagesProcessed < totalPagesToProcess) {
+            request.setPageNumber(pagesProcessed);
+            candidatePage = getSavedListCandidates(savedList, request);
+            List<Candidate> candidates = candidatePage.getContent();
+            processCandidateReassignment(candidates, newPartner);
+            entityManager.clear(); // Keeps candidates from piling up in persistence context
+            pagesProcessed++;
+        }
+    }
+
+    /**
+     * For each candidate on given list, sets partnerId on associated user object, saves to DB and
+     * updates the corresponding elasticsearch index entry.
+     * @param candidateList list of candidates
+     * @param newPartner the new partner to which they will be assigned
+     */
+    private void processCandidateReassignment(
+        List<Candidate> candidateList, Partner newPartner
+    ) {
+        for (Candidate candidate : candidateList) {
+            User candidateUser = candidate.getUser();
+            candidateUser.setPartner((PartnerImpl) newPartner);
```

Comment:
It may be worth wrapping this in an 'if (newPartner instance of PartnerImpl) {' check just in case. With some logging in the unexpected event that it isn't.

Reply:
👍 absolutely, will do.